File: godfather/roles/mafia/consigliere.py
from godfather.roles.mixins import SingleAction, MafiaMember
from godfather.game.types import Priority

DESCRIPTION = 'You may check one person for their exact role each night'


class Consigliere(MafiaMember, SingleAction):
    """
    A corrupted investigator who has been bribed to gather information for Mafia.

    - Win Condition: Kill anyone who will not submit to the Mafia.

    + Abilities: Check one person for their exact role each night.
    + If there are no mafia roles left who are capable of killing then you will become the goon.
    + You can talk with other mafia members.
    """
    name = 'Consigliere'
    description = DESCRIPTION

    def __init__(self):
        super().__init__()
        self.action = 'check'
        self.action_gerund = 'checking'
        self.action_priority = Priority.CONSIG
        self.action_text = 'check a player'
        self.categories.append('Mafia Support')

    async def tear_down(self, actions, player, target):
        await player.user.send(f'Your target must be a **{target.role.name}**.')

File: src/lib/mafia/roles/mafia/Consigliere.ts
import MafiaRole from '@mafia/mixins/MafiaRole';
import type Player from '@mafia/structures/Player';
import { RoleCheckAction } from '../../actions/common/RoleCheckAction';
import type { NightAction } from '../../managers/NightAction';
import { ActionRole } from '../../structures/ActionRole';

class Consigliere extends ActionRole {
	public name = 'Consigliere';
	public actions: NightAction[] = [new RoleCheckAction(this)];

	public constructor(player: Player) {
		super(player);
		this.description = this.game.t('roles/mafia:consigliereDescription');
	}
}

Consigliere.aliases = ['Consig'];
Consigliere.categories = [...Consigliere.categories, 'Mafia Support'];

export default MafiaRole(Consigliere);
